core/vm/dynamic_params.go

```diff
@@ -33,6 +33,14 @@ const (
 
 var getAddressForFuncABI, _ = abi.JSON(strings.NewReader(getAddressForABI))
 
+func GetRegisteredAddress(evm *EVM, registryId common.Hash) (common.Address, error) {
+	addr, err := GetRegisteredAddressWithEvm(registryId, evm)
+	if err != nil {
+		return common.ZeroAddress, nil
+	}
+	return *addr, nil
```

Comment:
Why the switch from pointer to non pointer return value here? I think the pointer approach is actually superior to the non pointer approach because a function that tries to use a nil pointer without checking it will panic, whereas a function that tries use a zero address without checking it, may succeed (E.G a send to a zero address will have no problem, the user might take issue with it though).


Also looking at the next [PR](#1462) in this chain of PRs I can see that the change caused significant code churn so that is another reason against making the change.

That is also a reason I think that these PRs would be better presented as a single PR split into nicely formed commits, since having them separated like this makes it difficult to assess the true impact of a change, and subsequently making a change here requires it to be propagated to the upstream PRs, which just adds a lot of overhead.

Reply:
+1 to making this one PR. The commit are already nicely formatted, so I've mainly been looking at the last PR and then looking at all changes from master on it.